### PR TITLE
Form validation

### PR DESCRIPTION
**Pull request by tesk9**
From your profile page (/#/user/:id -- get your id by looking for yourself in the bootstrap file or by looking for yourself in /user) you should see an "Edit" button. After you click this button, you should see a form filled with your profile info. 

Original form functionality should remain: 

If you edit this info and hit cancel, the form should disappear and your original data should reappear.

If you edit this info and hit submit, the form should disappear and your new values should show up. If you go to the /user/:your-id, you should now see your updated data.

Additional form functionality: validation. Submit button should be disabled when required fields are empty.

Validation has also been added to the form at the top of all team pages: /#/team/:teamId.

To test the last bit of validation, you will need to make yourself an admin of the core team (id=1). Then a form should appear at the bottom of /#/team/1. This form should not submit when empty.

### CODE REVIEW

**Comment by Conroyce** [comment]
tested, code looks good :+1:

File: assets/js/controllers/user_ctrl.js [hunk 1]
```diff
@@ -1,51 +1,61 @@
 app.controller("userCtrl", ["$scope", "$routeParams", "User", "AuthService", function($scope, $routeParams, User, AuthService){
```
Comment (jplouie) [inline assets/js/controllers/user_ctrl.js]: needs a space before curly brace

File: assets/js/controllers/user_ctrl.js [hunk 2]
```diff
@@ -1,51 +1,61 @@
...
-  $scope.update = function() {
-    var $elem = angular.element("#user-profile");
-    var $elems = $elem.children("input")
-
-    $scope.user.name = angular.element($elems[0]).val();
-    $scope.user.email = angular.element($elems[1]).val();
-    $scope.user.twitter = angular.element($elems[3]).val();
-    $scope.user.$update();
-    pageView();
-  }
```
Comment (jplouie) [inline assets/js/controllers/user_ctrl.js]: semicolon

File: assets/js/controllers/user_ctrl.js [hunk 3]
```diff
@@ -1,51 +1,61 @@
...
+  userRecord.$promise.then(function(user) {
+    setForm();
+    setUserPar();
+  });
+
   $scope.isOwn = false;
+  $scope.showForm = false;
   var currentUser;
   AuthService.currentUser(function(user) {
     $scope.isOwn = userId === user.id;
```
Comment (delasky) [inline assets/js/controllers/user_ctrl.js]: add parentheses

**Pull request merged**